### ISSUE TITLE
Opaque error stack in debugging script

### ISSUE THREAD

**Issue by Chanonry** [actions/github-script#112]
**Describe the bug**

Error returned is :

SyntaxError: Invalid or unexpected token
Error: Unhandled error: SyntaxError: Invalid or unexpected token

But the stack is not much help in identifying the source??

<img width="948" alt="Screenshot 2021-02-05 at 15 42 23" src="https://user-images.githubusercontent.com/20876829/107055107-da636800-67c8-11eb-9dd0-1e3d2bc13ef7.png">

**To Reproduce**

I am following a DemoDays Video with Peter Martin et al. Could not find the repo with the code so had to copy from the video.
https://www.youtube.com/watch?v=ojA5LM7ElTY

I have an Action that creates a GitHub Deployment. This deployment event triggers a second workflow to deploy a stack to aws with terraform. The Action creating the deployment succeeds and the second workflow is triggered successfully.

As part of this workflow there is a script that gets the deployment details for use with Terraform, it is this script that is failing.

Workflow below.

Expected behavior:

Data is extracted from the context and env vars are set.

Indentation may be a bit odd here but it is fine in the code...

    `
      name: Stack Deploy

    on:
      deployment

    jobs:
      create:
        # Only run if a deployment has been created
        if: github.event.action == 'created'

    runs-on: ubuntu-20.04

    env:
      TERRAFORM_VERSION: 0.14.5
      AWS_ACCESS_KEY_ID: ${{ secrets.NK_AWS_ACCESS_KEY_ID }}
      AWS_SECRET_ACCESS_KEY: ${{ secrets.NK_AWS_SECRET_ACCESS_KEY }}
      AWS_DEFAULT_REGION: ${{ secrets.NK_AWS_DEFAULT_REGION }}  
      GITHUB_PAT: ${{secrets.WRITE_PACKAGES_PAT}}  
      GITHUB_DEPLOYMENT_TOKEN: ${{ secrets.DEPLOYMENT_TOKEN }} 
      SSH_KEY: ${{secrets.NK_MODULES_SSH_KEY}} 
      ACTIONS_ALLOW_UNSECURE_COMMANDS: true

    steps:
      - name: Obtain Deployment Details for Terraform
        id: deployment
        uses: actions/github-script@v2
        with:
          github-token: ${{ env.GITHUB_DEPLOYMENT_TOKEN }}
          script: |
            console.log(JSON.stringify(context.payload, null, 2));

            const deployment = context.payload.deployment
              , deploymentPayload = JSON.parse(deployment.payload);
            ;
            console.log('Deployment Payload:');
            console.log(JSON.stringify(deploymentPayload));

            const statuses = await github.repos.listDeploymentStatuses({
              ...context.repo,
              deployment_id: deployment.id
            });

            # no statuses means pending
            if (statuses.data.length === 0) {
              // update the status to in progress , then start the terraform run
              await github.repos.createDeploymentStatus({
                ...context.repo,
                deployment_id: deployment.id,
                state: 'in_progress',
                required_contexts: [],
                mediaType: {previews: ['flash']}
              });

              // Export environment variables for Terraform to use
              core.exportVariable('TF_VAR_ENVIRONMENT', deploymentPayload.environment);
              core.exportVariable('TF_VAR_VERSION', deploymentPayload.version);
              core.exportVariable('TF_VAR_REGION', deploymentPayload.region);

              core.setOutput('deployment_id', deployment.id);
            } else {
              console.log(`unexpected state found on deployment: `);
              console.log(JSON.stringify(statuses.data[0].state, null , 2));

              core.setFailed(`Unexpected state '${statuses.data[0].state}' on deployment '${statuses.data[0].environment}'`};
            }

      - name: Check out source code
        uses: actions/checkout@v2

      - name: Set up SSH for git clone of private repo
        run: |
          mkdir ~/.ssh
          cp ${{env.SSH_KEY}} ~/.ssh/id_rsa
          chmod 700 ~/.ssh/
          chmod 600 ~/.ssh/id_rsa
          eval “$(ssh-agent -s)”
          ssh-add ~/.ssh/id_rsa

      # TERRAFORM STUFF ..................  

      # On success we need to update the deployment status
      - name: Report Success
        if: success()
        uses: actions/github-script@v2
        with:
          script: |
            const deployment = context.payload.deployment;

            await github.repos.createDeploymentStatus({
                ...context.repo,
                deployment_id: deployment.id,
                state: 'success',
                log_url: `https://github.com/${process.env.GITHUB_REPOSITORY}/actions/runs/${process.env.GITHUB_RUN_ID}`,
                mediaType: {
                  previews: ['ant-man']},
            });

      # On FAILURE we need to update the deployment status
      - name: Report Failure
        if: failure()
        uses: actions/github-script@v2
        with:
          script: |
            const deployment = context.payload.deployment;

            await github.repos.createDeploymentStatus({
                ...context.repo,
                deployment_id: deployment.id,
                state: 'failure',
                required_contexts: [],
                mediaType: {previews: ['flash']},
            });

`

**Comment by Chanonry** [comment]
Thanks for your help, not

**Comment by yi-Xu-0100** [comment]
I have the same error when I want to make a PR.

```bash
Run actions/github-script@v3.1.0
  with:
    script: await github.pulls.create({
    owner: context.repo.owner,
    repo: context.repo.repo,
    head: prepare-release,
    base: main,
    title: 'chore(release): :bookmark: release v1.0.4',
    body: '
  ### Bug Fixes
  
  - **actions:** 🐛 fix not found branch ([25b4e19](https://github.com/yi-Xu-0100/test/commit/25b4e1926a26dca753a2b6bb8627881f47941428))
  - fix push error ([62458bf](https://github.com/yi-Xu-0100/test/commit/62458bf2fede7fe842cd9176cec6920a339469d8))',
    maintainer_can_modify: true,
  })
  
    github-token: ***
    debug: false
    user-agent: actions/github-script
    result-encoding: json
  env:
    BASE_BRANCH: main
    HEAD_BRANCH: prepare-release
SyntaxError: Invalid or unexpected token
    at new AsyncFunction (<anonymous>)
    at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:5368:56)
Error: Unhandled error: SyntaxError: Invalid or unexpected token
    at main (/home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:5394:26)
    at Module.720 (/home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:5378:1)
    at __webpack_require__ (/home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:24:31)
    at startup (/home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:43:19)
    at /home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:49:18
    at Object.<anonymous> (/home/runner/work/_actions/actions/github-script/v3.1.0/dist/index.js:52:10)
    at Module._compile (internal/modules/cjs/loader.js:959:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)
```

**Comment by brandonkrull** [comment]
I encountered similar errors and it MIGHT be because of your comment syntax (`# no statuses means pending`). since that's actually in the `script` block, it's expecting comments to be denoted using js syntax, i.e. `//`.
